zebrad/src/commands/start/sync.rs

```diff
@@ -141,12 +141,17 @@ where
             }
         }
 
+        self.prospective_tips
+            .retain(|tip| !download_set.contains(tip));
```

Comment:
I think this will remove all tips, since the tips are queued for download on line 124 above

Reply:
new_tip is removed from the `hashes` list before the hashes are pushed into `download_set`, so line 124 should be queueing everything _except_ the tip

Reply:
Where does the tip get downloaded, then?

Reply:
Ah, I see, that's why there's a new request_blocks invocation below.